### ISSUE TITLE
fatal error: google/protobuf/inlined_string_field.h: No such file or directory  #include <google/protobuf/inlined_string_field.h>

### ISSUE THREAD

**Issue by roshan-dongre** [tensorflow/tensorflow#22320]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: Commit cac963862be3faa421c559f39033c9bfb3b27a51
- **Python version**: 2.7
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

After following this tutorial: https://tuatini.me/building-tensorflow-as-a-standalone-project/ and running the following command:
g++ -std=c++11 -Wl,-rpath='$ORIGIN/lib' -Iinclude -Llib main.cpp -ltensorflow_cc -o exec

I'm getting the error "This file was generated by a newer version of protoc which is incompatible with your Protocol Buffer headers. Please update" and "google/protobuf/inlined_string_field.h: No such file or directory  #include <google/protobuf/inlined_string_field.h>" 

After looking through the commits I think the protobuf url is outdated.

![screenshot from 2018-09-17 12 39 06](https://user-images.githubusercontent.com/14967965/45640040-2cc85080-ba77-11e8-90e6-95fa13d909cd.png)

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by roshan-dongre** [comment]
~/Desktop/tensorflow/tensorflow/contrib/makefile/downloads/protobuf/src protoc --version gives libprotoc 3.5.0 but inlined_string_field.h is not in 3.5.0

**Comment by roshan-dongre** [comment]
Figured out a way around this and was able to run a custom c++ program with g++ outside the tensorflow repo without using bazel other than for building the .so files. I'll be posting a tutorial soon for anyone who needs it.

**Comment by dhingratul** [comment]
+1

**Comment by dhingratul** [comment]
> /Desktop/tensorflow/tensorflow/contrib/makefile/downloads/protobuf/src protoc --version gives libprotoc 3.5.0 but inlined_string_field.h is not in 3.5.0 (generated after running build_all_linux.sh)

workspace.bzl downloads 3.6.0, i am trying to hardcode and revert to 3.5.0, have you tried that, does that work ?

**Comment by roshan-dongre** [comment]
Yeah you can do that or you can get the 3.6.0 protobufs from the protobuf github repo which is what I ended up doing.

**Comment by harshini-gadige** [comment]
@roshan-dongre Hi, can this issue be closed as you figured out a way to resolve this ?

**Comment by achalshah20** [comment]
I changed protobuf version here as 3.6.0 here and getting the same error and op? 

![image](https://user-images.githubusercontent.com/9772589/45911631-e0f50e80-bdca-11e8-916a-4efe25f58ce0.png)

**Comment by achalshah20** [comment]
Hi @roshan-dongre , Can you explain where exactly did you copy absl folder for fixing #22240 (Path for external folder) ? 

" I was able to fix the absl issue by pulling and copying over the absl folder to the external folder "

**Comment by SamuelLarkin** [comment]
Hi,
  I'm facing a version issue trying to compile tensorflow-1.11.0 in order to use it in a C++ project.  I think the issue is that bazel use protobuf-3.6.0 in `tensorflow/workspace.bzl` but the `tensorflow/contrib/makefile/download_dependencies.sh` fetch version 3.5.0 thus when you follow the [instructions](https://tuatini.me/building-tensorflow-as-a-standalone-project/) where you copy the proto stuff in Tensorflow's install_dir, we get a version mismatch.
